pkg/controller/registry/resolver/cache/operators.go

```diff
@@ -158,6 +164,9 @@ type Entry struct {
 	SourceInfo   *OperatorSourceInfo
 	Properties   []*api.Property
 	BundlePath   string
+	// Since there are separate cache entries for the same bundle in different channels,
+	// This should contain at most one deprecation for package, channel and bundle.
+	Deprecations []Deprecation
```

Comment:
Two things:
1. An entry is already specific to a package, channel, and bundle version, right? (looking at `SourceInfo`, that seems to be the case). Given that, it seems like we could make `Deprecations` be a single struct that has pointer fields for `Package`, `Channel`, and `Bundle` `api.Deprecation` structs, and then we don't need strings to encode `Type` and `Name`?
2. Would it make sense to put this inside the `OperatorSourceInfo` struct, since that already seems to have information related the package and channel catalog metadata?

Reply:
Thanks for suggesting this, I think it cleans things up a lot, particularly when constructing the conditions and detecting required changes.